src/server/raftkv2/mod.rs

```diff
@@ -153,18 +153,21 @@ impl<EK: KvEngine, ER: RaftEngine> tikv_kv::Engine for RaftKv2<EK, ER> {
 
         let mut header = new_request_header(ctx.pb_ctx);
         let mut flags = 0;
-        if ctx.pb_ctx.get_stale_read() && ctx.start_ts.map_or(true, |ts| !ts.is_zero()) {
-            let mut data = [0u8; 8];
-            (&mut data[..])
-                .encode_u64(ctx.start_ts.unwrap_or_default().into_inner())
-                .unwrap();
+        if ctx.pb_ctx.get_stale_read() {
```

Comment:
In the previous, the flag of stale_read will not be set if `start_ts` is 0.

Reply:
Previously, `start_ts` encoded in the flag data will only be used for the stale read. Now it's needed to check the flashback ts for each request, so it's ok to break this logic.

Reply:
So need to resolve the situation that due to `start_ts` is 0, we get `stale read` flag but data is empty. I found that`decode_u64` checks data size. https://github.com/tikv/tikv/blob/master/components/raftstore/src/store/worker/read.rs#L970